### ISSUE TITLE
Gradle logging hygiene

### ISSUE THREAD

**Issue by rmorrise** [gradle/gradle#2688]
Implement log hygiene in Gradle's default configuration, as described in:
https://discuss.gradle.org/t/gradle-daemon-produces-a-lot-of-logs/9905

### Expected Behavior

Gradle should be configured out-of-the-box with a rolling file appender or similar mechanism so that the daemon logs are periodically cleaned up.

### Current Behavior

Gradle produces daemon logs in a hidden user directory. These logs accumulate over time and are never cleaned up.

### Context

My Grails build was leaking daemon processes, contributing to low disk space on C:
(The daemon process leak is beyond the scope of this feature request.)

### Steps to Reproduce (for bugs)

Not a bug.

### Your Environment

 * Build scan URL: 

Win7 x64, Grails 3.3.0, Gradle 3.4.1, Groovy 2.4.7, JVM 1.8.0_66

**Comment by staffanf** [comment]
I just found 20 Gb worth of deamon logs in ~/.gradle. Some of these were 4Gb full of System.out output some debug task I ran some month ago.
Would be nice with some kind of log rotation at least.

**Comment by agilob** [comment]
I just removed 89Gb of text files in `.gradle/daemon/4.7` -.-'

**Comment by lostcart** [comment]
I just deleted 160GB after Android studio complained I didn't have enough disk space - 1 day later and log files are at 2.4GB - this is insane . (I also have over 9GB of memory used from AS, but that's another issue :D)

**Comment by fbe** [comment]
+1

**Comment by petarmarijanovicfive** [comment]
+1

I deleted 100+GB. 

This is a big issue. Gradle shouldn't log anything to a separate file unless it is specifically configured to do so.

**Comment by coderkun** [comment]
It looks like the log level in these files is `DEBUG`. If it is possible to configure the log level the Gradle daemon is using, it will reduce the amount of logs greatly.

**Comment by chaosky** [comment]
> It looks like the log level in these files is `DEBUG`. If it is possible to configure the log level the Gradle daemon is using, it will reduce the amount of logs greatly.

It looks like it's hardcoded in [DaemonMain.java#L192](https://github.com/gradle/gradle/blob/v5.6.4/subprojects/launcher/src/main/java/org/gradle/launcher/daemon/bootstrap/DaemonMain.java#L192), probably providing a `System.property` such as `-Dorg.gradle.daemon.logging.level=(quiet,warn,lifecycle,info,debug)` to control it.

@fbe @rmorrise

**Comment by lingyfh** [comment]
org.gradle.logging.level=(quiet,warn,lifecycle,info,debug)

[gradle Build Environment config url](https://docs.gradle.org/current/userguide/build_environment.html)
[choosing_a_log_level](https://docs.gradle.org/current/userguide/logging.html#sec:choosing_a_log_level)

**Comment by kevinstembridge** [comment]
+1

**Comment by brokenalarms** [comment]
Any progress on this? This fills my hard drive and blocks work without warning until I investigate and fix each time. Would have considered this a major issue due to it impacting the entire user's system as a side effect.

**Comment by JacquesLeRoux** [comment]
The problem is that none of the log levels is able to render nothing. I don't want to have to clean by hand my production VM every now an then :/ 
I'd like an option to mute the daemon log!

**Comment by bugs84** [comment]
It would be great if there was some "good default" like "delete older than a month". "Do not eat up the whole disk by default"
Imagine: My colleague setup builds. And after some time someone else has to solve "Why disk is full and everything is broken :("
(and someone else knows nothing about Gradle)

**Comment by spamdr** [comment]
+1

**Comment by cyberpwnn** [comment]
+1

**Comment by TigerPhenix** [comment]
+1

**Comment by markrmiller** [comment]
+1 - jesus this is insane and I can't believe it's still going.

**Comment by chylex** [comment]
Any updates on this? I just deleted several GB of not just logs, but also some so-called "temporary" files Gradle never cleaned up..

**Comment by p4paul** [comment]
+1 
Just removed 3.1GB of logs with `find ~/.gradle -name *.out.log -exec rm {} \;`

**Comment by charlesritchea** [comment]
This doesn't have to be automatic, just a command like `docker system prune`

**Comment by JacquesLeRoux** [comment]
Not everybody use docker, OK can be any such command, but better be safe than sorry with an automated option. OK could be a any such command runs by a cron job. But sincerely it would much more easy with an embedded documented function for that rather than being surprised one day with a disk full!

**Comment by winterbe** [comment]
We just deleted 60+GB on our CI agents after figuring this out, multiple hundreds of GB in total space. We don't use docker, so we have to setup a cleanup job I guess. An easy build-in option to prune logs is highly appreciated. I don't think many people outside of the Gradle team even look at those logs.

**Comment by slPerryRhodan** [comment]
Default tasks for cleaning the logs and cache would be very nice.

**Comment by tkrah** [comment]
If this is going to be fixed - please also have a look in the "/tmp" directory for the kotlin-daemon log files (kotlin daemon is spawned by gradle if kotlin is used e.g. in buildSrc), those are gathering there too:

/tmp/kotlin-daemon.2022-07-11.10-04-03-480.00.log

and so one for each daemon and each day - on CI this is gathering up fast.

**Comment by ebrandsberg** [comment]
15GB of logs here... All I can say is wtf?  No production code should have this type of log issue today, and when opened up how many years ago?  To still be a problem?

**Comment by nhouser9** [comment]
it's fully ridiculous that this isn't fixed. just found out this was taking up 20% of my SSD. ideal would be providing a default that makes any sense. in lieu of that, please at least make this configurable.

**Comment by pwagland** [comment]
We just ran across this same issue, a month into building, on our CI server we had 40G of logs in this directory, almost a terabyte across the CI landscape…

**Comment by jbartok** [comment]
Thank you for your interest in Gradle!

This feature request is in the backlog of the relevant team and is prioritized by them.

**Comment by JacquesLeRoux** [comment]
Good news, thanks @jbartok :)

**Comment by sreich** [comment]
this bug just killed my build server...

rule #1 of ever writing code using a temp dir is the owning app must clean itself up...if it doesn't, then it's broken. i'm shocked to see that it's been so many years, going back many gradle versions, and this has still been an issue...but at least it's getting attention now

in my case, it is `kotlin-daemon.log`, so those will hopefully be addressed in the upcoming fix...but it looks like there are also junit files and other things in the tmpdir that it may not be cleaning up (cannot confirm that bit)

until then, looks like we're going to have to write a cleanup job hack for this 1 application so it doesn't break the server every month

**Comment by tanranran** [comment]
please Request repair, I am especially afraid that the computer will break down

**Comment by davidgiga1993** [comment]
I've open a MR which should resolve this issue, note that the default log level is still debug (but can be configured now) as I'm waiting on feedback of this MR.

**Comment by eskatos** [comment]
Gradle 9.4.0 will delete daemon log files older than two weeks.
This is not configurable but should provide sane defaults.

* See https://github.com/gradle/gradle/pull/35765
